include/iarc7_vision/OpticalFlowEstimator.hpp

```diff
@@ -0,0 +1,213 @@
+#ifndef IARC7_VISION_OPTICAL_FLOW_ESTIMATOR_HPP_
+#define IARC7_VISION_OPTICAL_FLOW_ESTIMATOR_HPP_
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/gpu/gpu.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <ros/ros.h>
+
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <ros_utils/SafeTransformWrapper.hpp>
+
+#include <geometry_msgs/TwistWithCovarianceStamped.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/image_encodings.h>
+
+namespace iarc7_vision {
+
+struct OpticalFlowEstimatorSettings {
+    double fov;
+    double min_estimation_altitude;
+    int win_size;
+    int max_level;
+    int iters;
+    int points;
+    double quality_level;
+    int min_dist;
+    double scale_factor;
+    double variance;
+    double variance_scale;
+    double x_cutoff_region_velocity_measurement;
+    double y_cutoff_region_velocity_measurement;
+    int debug_frameskip;
+    double tf_timeout;
+};
+
+struct OpticalFlowDebugSettings {
+    bool debug_vectors_image;
+    bool debug_average_vector_image;
+    bool debug_times;
+};
+
+class OpticalFlowEstimator {
+  public:
+
+    //////////////////
+    // CONSTRUCTORS //
+    //////////////////
+
+    OpticalFlowEstimator(
+            const OpticalFlowEstimatorSettings& flow_estimator_settings,
+            const OpticalFlowDebugSettings& debug_settings);
+
+    ////////////////////
+    // PUBLIC METHODS //
+    ////////////////////
+
+    /// MUST be called when either of the settings objects passed into the
+    /// constructor have their variables changed
+    bool __attribute__((warn_unused_result)) onSettingsChanged();
```

Comment:
Same thing, make these different names

Reply:
Make what different names? There should only be one `onSettingsChanged` function

Reply:
You're right, I misread that line